### ISSUE TITLE
5.0.0 Release

### ISSUE THREAD

**Issue by sunshinejr** [sunshinejr/Moya-ModelMapper#37]
I would love to do 5.0.0 release in a near future. This would be almost just after the Moya 9.0.0 release. It should include:

- [x] SPM compatibility
- [x] Carthage compatibility
- [x] Better method naming
- [x] Moya 9 compatibility
- [x] If possible Swift 3.1/3.2/4.0 compatibility (probably 4.0 right now without Rx/ReactiveSwift)
- [x] Better error handling #38

**Comment by sunshinejr** [comment]
I've updated lots of stuff in there! Right now it is building & publishing to CocoaPods version `5.0.0-beta.1`. Once its up, would you be up for checking SPM/Carthage compatibility, @josefdolezal? Would be awesome. 🎉

**Comment by josefdolezal** [comment]
I am out until tomorrow, but I will definetely check both of it.

It looks like Moya 9.0 will use new format of Package manifest (Moya/Moya#1240). Maybe we should also update to V4 syntax (which is Xcode 9 compatible). It will allow us to use "subspecs" for reactive extensions targets.

I can also handle the PR with manifest update for Moya.

**Comment by sunshinejr** [comment]
Great, do it whenever you have free time available. Thanks, @josefdolezal!

**Comment by josefdolezal** [comment]
Hey @sunshinejr, sorry for the delay! My Macbook decided not to work anymore so I had to take for repaire and setup my backup Mac..

However, I checked the changes on my iPad and it looks awesome! I haven’t checked it into details, but it looks like the guide for Swift PM is incorrect. It shows syntax for V3 but with Package initializer which does not really exists.

I will create PR with fix and check the Carthage compatibility once my backup mac is ready (which should be later this week).

**Comment by sunshinejr** [comment]
Don't worry @josefdolezal, take your time :) Thanks for the info!

**Comment by josefdolezal** [comment]
I am in the middle of updating Swift PM and Carthage to work properly with Moya 9.0. The Readme also has to be updated to reflect the changes. However, for Swift PM, we have to make new release (V3 PackageDescription does not support branch/revision references). What should I put in the docs?

```swift
Version(5, 0, 0, prereleaseIdentifiers: ["beta", "1"])
// OR
Version(5, 0, 0, prereleaseIdentifiers: ["beta", "2"]
```

None of it seems right since neither of it works (beta.2 will work right after the release). We can also move to PackageDescription4 but we would have to drop Xcode 8/Swift 3.1 support. What do you think about that?

**Comment by sunshinejr** [comment]
Hmm, if I delete current beta tag and create a new one after your PR, it would probably work with beta.1, wouldn't it?

**Comment by josefdolezal** [comment]
Yup, this should solve my dilemma 😅

**Comment by sunshinejr** [comment]
Jus tested all 3 package managers: SPM, Carthage and CocoaPods, and it's all working good! Thanks @josefdolezal for helping in this one, really appreciate it! 🎉

**Comment by josefdolezal** [comment]
Awesome, glad to hear that! Please let me know if there is any other thing I can help you with.

**Comment by sunshinejr** [comment]
I will release 5.0.0 in a while, because I just released Moya 9.0.0. So it would be awesome if you could check if everything works correctly for you. 

Also, I would love a script to check package managers compatibility - checking if SPM/Carthage/CocoaPods works correctly. Do you happen to know if something like this exists, @josefdolezal? 🤔

**Comment by sunshinejr** [comment]
5.0.0 released! 🎉 🎉 🎉
